### PR TITLE
[FSSDK-9781] warn on duplicate experiment key

### PR DESCRIPTION
**Pull request by andrewleap-optimizely**
## Summary
- Ensure newest duplicate experiment key takes precedence (current behavior)
- log a warning when a duplicate experiment key is found

## Test plan
- added test for dupe experiment

## Ticket
[FSSDK-9781](https://jira.sso.episerver.net/browse/FSSDK-9781)

### CODE REVIEW

File: lib/optimizely/optimizely_config.rb [hunk 1]
```diff
@@ -91,6 +92,7 @@ def audiences_map
 
     def experiments_map
       experiments_id_map.values.reduce({}) do |experiments_key_map, experiment|
+        @logger.log(Logger::WARN, "Duplicate experiment keys found in datafile: #{experiment['key']}") if experiments_key_map.key? experiment['key']
```
Comment (jaeopt) [inline lib/optimizely/optimizely_config.rb:95]: Looks good. A question is the order in datafile is managed in experiment_id_map. Just for consistency, we're supposed to return the last experiment in the datafile.

**Review by jaeopt — CHANGES_REQUESTED** [review]
Looks good. A clarification.

File: lib/optimizely/optimizely_config.rb (hunk 1 above)
Comment (andrewleap-optimizely) [reply lib/optimizely/optimizely_config.rb:95]: Yea Ruby iterates over hash values in insertion order, so the last instance of an experiment key will take precedence.

**Review by jaeopt — APPROVED** [review]
LGTM

**Pull request merged**